File: GrooveOn.Models/ResponseObjects/GenreResponse.cs
using System;

namespace GrooveOn.Model.ResponseObjects
{
    public class GenreResponse
    {
        public int Id { get; set; }
        public string ExternalGenreId { get; set; } = string.Empty;
        public string Source { get; set; } = string.Empty;
        public string Name { get; set; } = string.Empty;
        public DateTime CreatedAt { get; set; }
    }
}

File: UI/grooveon_desktop/lib/models/response/genre_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'genre_response.g.dart';

@JsonSerializable()
class GenreResponse {
  final int id;
  final String externalGenreId;
  final String source;
  final String name;
  final DateTime? createdAt;

  GenreResponse({
    required this.id,
    required this.externalGenreId,
    required this.source,
    required this.name,
    this.createdAt,
  });

  factory GenreResponse.fromJson(Map<String, dynamic> json) =>
      _$GenreResponseFromJson(json);

  Map<String, dynamic> toJson() => _$GenreResponseToJson(this);
}